core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreRegistryPersistenceManager.java

```diff
@@ -0,0 +1,373 @@
+                keyStoreMetadata.setPublicCert((byte[]) pubKeyResource.getContent());
+            }
+        }
+        return keyStoreMetadata;
+    }
+
+    /**
+     * Check whether a key store exists in the registry.
+     *
+     * @param keyStoreName Key store name.
+     * @return true if the key store exists, false otherwise.
+     * @throws SecurityException If an error occurs while checking the existence of the key store.
+     */
+    public boolean isKeyStoreExists(String keyStoreName) throws SecurityException {
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        try {
+            return registry.resourceExists(path);
+        } catch (RegistryException e) {
+            throw new SecurityException("Error checking the existence of key store : " + keyStoreName, e);
+        }
+    }
+
+    /**
+     * Get the last modified date of the key store.
+     *
+     * @param keyStoreName Key store name.
+     * @return Last modified date of the key store.
+     */
+    public Date getKeyStoreLastModifiedDate(String keyStoreName) {
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        try {
+            return registry.get(path).getLastModified();
+        } catch (RegistryException e) {
+            LOG.error("Error while getting the last modified date of the key store: " + keyStoreName, e);
+            return null;
+        }
+    }
+
+    /**
+     * Get the key store password for the given key store name.
+     *
+     * @param keyStoreName Key store name.
+     * @return Encrypted password of the key store.
+     * @throws SecurityException If there is an error while getting the key store password.
+     */
+    public String getEncryptedKeyStorePassword(String keyStoreName) throws SecurityException {
+
+        String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+        try {
+            if (registry.resourceExists(path)) {
+                Resource resource = registry.get(path);
+                return resource.getProperty(RegistryResources.SecurityManagement.PROP_PASSWORD);
+            } else {
+                throw new SecurityException("Key Store with a name : " + keyStoreName + " does not exist.");
+            }
+        } catch (RegistryException e) {
+            throw new SecurityException("Error getting password for key store : " + keyStoreName, e);
+        }
+    }
+
+    /**
+     * Get the encrypted private key password for the given key store name.
+     *
+     * @param keyStoreName Key store name.
+     * @return Encrypted private key password.
+     * @throws SecurityException If an error occurs while getting the private key password.
+     */
+    public String getEncryptedPrivateKeyPassword(String keyStoreName) throws SecurityException {
```

Comment:
Shall we modify the method to return the decrypted value?

Reply:
Fixed in the latest changes